core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -466,7 +466,7 @@ object SparkSubmit extends CommandLineUtils {
       // Other options
       OptionAssigner(args.executorCores, STANDALONE | YARN, ALL_DEPLOY_MODES,
         sysProp = "spark.executor.cores"),
-      OptionAssigner(args.executorMemory, STANDALONE | MESOS | YARN, ALL_DEPLOY_MODES,
+      OptionAssigner(args.executorMemory, ALL_CLUSTER_MGRS, ALL_DEPLOY_MODES,
```

Comment:
also we're talking about a net addition of 7 LOC in `SparkContext.scala`, about half of which are comments and warning logs. It's really not that much code.